Content.Shared/Destructible/SharedDestructibleSystem.cs

```diff
@@ -26,7 +28,7 @@ public void BreakEntity(EntityUid owner)
 /// <summary>
 ///     Raised when entity is destroyed and about to be deleted.
 /// </summary>
-public sealed class DestructionEventArgs : EntityEventArgs
+public sealed class DestructionEventArgs : CancellableEntityEventArgs
```

Comment:
instead of destruction event args being cancellable, you should add a new cancellable event that is raised before this one. With the present logic, you can only cancel the deletion, and not the actual effects of DestructionEventArgs, which is probably unwanted.